tide/master/ScreenControllerFactory.cpp

```diff
@@ -40,30 +40,70 @@
 #include "ScreenControllerFactory.h"
 
 #include "MultiScreenController.h"
-#include "PlanarController.h"
 
+#include <QDebug>
 #include <QStringList>
 
+namespace
+{
+PlanarController::Type _getType(const QString& name = "Matrix")
+{
+    if (name == "UR9850")
+        return PlanarController::Type::TV_UR9850;
+    if (name == "UR9851")
+        return PlanarController::Type::TV_UR9851;
+    return PlanarController::Type::Matrix;
+}
+}
+
 std::unique_ptr<ScreenController> ScreenControllerFactory::create(
     const QString& ports)
 {
-    auto serialports = ports.split(';');
-    if (serialports.length() == 0)
+    const auto connections = processInputString(ports);
+    if (connections.size() == 0)
         return nullptr;
-    if (serialports.length() == 1)
-    {
-        return std::unique_ptr<ScreenController>(
-            new PlanarController(serialports[0],
-                                 PlanarController::Type::Matrix));
-    }
 
-    std::vector<std::unique_ptr<PlanarController>> controllers;
-    for (const auto serialport : serialports)
+    std::vector<std::unique_ptr<ScreenController>> controllers;
+    for (const auto key : connections.keys())
     {
-        auto controller = std::unique_ptr<PlanarController>(
-            new PlanarController(serialport, PlanarController::Type::TV));
-        controllers.push_back(std::move(controller));
+        controllers.emplace_back(
+            new PlanarController(key, connections.value(key)));
     }
+    if (controllers.size() == 1)
+        return std::move(controllers[0]);
+
     return std::unique_ptr<ScreenController>(
         new MultiScreenController(std::move(controllers)));
 }
+
+QMap<QString, PlanarController::Type>
+    ScreenControllerFactory::processInputString(const QString& ports)
+{
+    QMap<QString, PlanarController::Type> map;
```

Comment:
use a typedef for QMap<QString, PlanarController::Type>? It's quite long and repeated a lot of times.